eight/src/embedded/language/lexer.rs

```diff
@@ -124,3 +124,62 @@ pub(super) fn lex(source: String) -> Vec<Vec<Token>> {
 
     lexer.collect()
 }
+
+#[cfg(test)]
```

Comment:
It would be better to move this test to `embedded/language/tests.rs` to make file more readable. You can see an example in `client/` folder:

https://github.com/meppu/eight/blob/main/eight/src/client/mod.rs#L12-L13
https://github.com/meppu/eight/blob/main/eight/src/client/tests.rs